velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -260,5 +260,28 @@ TEST_F(DateTimeFunctionsTest, dateAdd) {
   EXPECT_EQ(parseDate("1969-12-31"), dateAdd(parseDate("5881580-07-11"), kMin));
 }
 
+TEST_F(DateTimeFunctionsTest, dateDiffDate) {
+  const auto dateDiff = [&](std::optional<int32_t> date1,
+                            std::optional<int32_t> date2) {
+    return evaluateOnce<int32_t, int32_t>(
+        "date_diff(c0, c1)", {date1, date2}, {DATE(), DATE()});
+  };
+
+  // Check null behaviors
+  EXPECT_EQ(std::nullopt, dateDiff(1, std::nullopt));
+  EXPECT_EQ(std::nullopt, dateDiff(std::nullopt, 0));
+
+  // Simple tests
+  EXPECT_EQ(-1, dateDiff(parseDate("2019-02-28"), parseDate("2019-03-01")));
+  EXPECT_EQ(-358, dateDiff(parseDate("2019-02-28"), parseDate("2020-02-21")));
+
+  // Account for the last day of a year-month
+  EXPECT_EQ(395, dateDiff(parseDate("2020-02-29"), parseDate("2019-01-30")));
+
+  // Check Large date
+  EXPECT_EQ(
+      -737790, dateDiff(parseDate("2020-02-29"), parseDate("4040-02-29")));
```

Comment:
does it support negative years? Could add a test as well.

Reply:
Just Updated.